mithril-aggregator/src/store/protocol_parameters_store.rs

```diff
@@ -20,193 +20,150 @@ pub trait ProtocolParametersStorer: Sync + Send {
             .await
             .unwrap();
 
-        assert!(res.is_none());
+        assert!(protocol_parameters_previous.is_none());
     }
 
     #[tokio::test]
-    async fn update_protocol_parameters_in_store() {
-        let protocol_parameters = setup_protocol_parameters(2);
-        let store = init_store(1, None);
-        let res = store
-            .save_protocol_parameters(protocol_parameters[0].0, protocol_parameters[1].1.clone())
+    async fn test_save_protocol_parameters_already_exist() {
+        let protocol_parameters = fake_data::protocol_parameters();
+        let epoch = Epoch(1);
+        let store = FakeProtocolParametersStorer::new(vec![(epoch, protocol_parameters.clone())]);
+        let protocol_parameters_previous = store
+            .save_protocol_parameters(epoch, protocol_parameters.clone())
             .await
             .unwrap();
 
-        assert_eq!(protocol_parameters[0].1, res.unwrap());
+        assert_eq!(Some(protocol_parameters), protocol_parameters_previous);
     }
 
     #[tokio::test]
-    async fn get_protocol_parameters_for_empty_epoch() {
-        let store = init_store(2, None);
-        let res = store.get_protocol_parameters(Epoch(0)).await.unwrap();
+    async fn test_get_protocol_parameters_exist() {
+        let protocol_parameters = fake_data::protocol_parameters();
+        let epoch = Epoch(1);
+        let store = FakeProtocolParametersStorer::new(vec![(epoch, protocol_parameters.clone())]);
+        let protocol_parameters_stored = store.get_protocol_parameters(epoch).await.unwrap();
 
-        assert!(res.is_none());
+        assert_eq!(Some(protocol_parameters), protocol_parameters_stored);
     }
 
     #[tokio::test]
-    async fn get_protocol_parameters_for_existing_epoch() {
-        let store = init_store(2, None);
-        let res = store.get_protocol_parameters(Epoch(1)).await.unwrap();
+    async fn test_get_protocol_parameters_do_not_exist() {
+        let protocol_parameters = fake_data::protocol_parameters();
+        let epoch = Epoch(1);
+        let store = FakeProtocolParametersStorer::new(vec![(epoch, protocol_parameters.clone())]);
+        let protocol_parameters_stored = store.get_protocol_parameters(epoch + 1).await.unwrap();
 
-        assert!(res.is_some());
+        assert!(protocol_parameters_stored.is_none());
     }
 
     #[tokio::test]
-    async fn test_retention_length() {
-        let store = init_store(2, Some(2));
-        let protocol_parameters = setup_protocol_parameters(3);
-        let _ = store
-            .save_protocol_parameters(Epoch(3), protocol_parameters[2].1.clone())
-            .await
-            .unwrap();
-        assert!(store
-            .get_protocol_parameters(Epoch(1))
-            .await
-            .unwrap()
-            .is_some());
-        let res = store.get_protocol_parameters(Epoch(2)).await.unwrap();
-        assert!(res.is_some());
-    }
+    async fn test_handle_discrepancies_at_startup() {
```

Comment:
We can add some explanation in the test name to understand what we are checking.
I suggest something like: test_handle_discrepandies_at_startup_should_complete_at_least_two_epochs

Reply:
Done :heavy_check_mark: